### PR TITLE
Add full type name for Lucene lock type

### PR DESCRIPTION
**Pull request by DrewScoggins**
In net9.0 the runtime now exposes it's own `Lock` type, https://github.com/dotnet/runtime/pull/87672. This causes a build break as there is a collision between the Lucene type and the runtime type.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs [hunk 1]
```diff
@@ -395,7 +395,7 @@ private async Task WriteAsync(string indexName, Action<IndexWriter> action, bool
                         var config = new IndexWriterConfig(LuceneSettings.DefaultVersion, analyzer)
                         {
                             OpenMode = OpenMode.CREATE_OR_APPEND,
-                            WriteLockTimeout = Lock.LOCK_POLL_INTERVAL * 3
+                            WriteLockTimeout = Lucene.Net.Store.Lock.LOCK_POLL_INTERVAL * 3
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:398]: I think this is the required change, please revert the other changes

**Review by hishamco — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs (hunk 1 above)
Comment (DrewScoggins) [reply src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:398]: Sorry about that, not sure how those other commits got in there. I will fix this up.

**Comment by DrewScoggins** [comment]
I added an alias for `Lucene.Net.Store` because that was colliding with `OrchardCore.Search.Lucene`. I am happy to do it another way if there is a better one, but this was what I came up with.

**Comment by DrewScoggins** [comment]
Does this change look good?

**Comment by hishamco** [comment]
I think the change should be in lock only

**Comment by DrewScoggins** [comment]
If the change is limited to just the one that you requested you can't build the project and get this error
```
D:\git\OrchardCore\src\OrchardCore.Modules\OrchardCore.Search.Lucene\Services\LuceneIndexManager.cs(398,48): error CS02
34: The type or namespace name 'Net' does not exist in the namespace 'OrchardCore.Search.Lucene' (are you missing an as
sembly reference?) [D:\git\OrchardCore\src\OrchardCore.Modules\OrchardCore.Search.Lucene\OrchardCore.Search.Lucene.cspr
oj::TargetFramework=net8.0]
```

This is why I made the change to add the alias for the `Lucene.Net.Store` namespace. If there is a better way to solve that issue I am happy to use that, but this is the only way that I know.

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 using Lucene.Net.Search;
 using Lucene.Net.Spatial.Prefix;
 using Lucene.Net.Spatial.Prefix.Tree;
-using Lucene.Net.Store;
+using LStore = Lucene.Net.Store;
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:14]:
```suggestion
using Lucene.Net.Store;
using LuceneLock = Lucene.Net.Store.Lock;
```

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs [hunk 3]
```diff
@@ -362,7 +362,7 @@ private Document CreateLuceneDocument(DocumentIndex documentIndex, LuceneIndexSe
             return doc;
         }
 
-        private BaseDirectory CreateDirectory(string indexName)
+        private LStore.BaseDirectory CreateDirectory(string indexName)
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:365]:
```suggestion
        private BaseDirectory CreateDirectory(string indexName)
```

Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:365]:
```suggestion
        private BaseDirectory CreateDirectory(string indexName)
```

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs [hunk 4]
```diff
@@ -428,7 +428,7 @@ private IndexReaderPool.IndexReaderLease GetReader(string indexName)
             var pool = _indexPools.GetOrAdd(indexName, n =>
             {
                 var path = new DirectoryInfo(PathExtensions.Combine(_rootPath, indexName));
-                var reader = DirectoryReader.Open(FSDirectory.Open(path));
+                var reader = DirectoryReader.Open(LStore.FSDirectory.Open(path));
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:431]:
```suggestion
                var reader = DirectoryReader.Open(FSDirectory.Open(path));
```

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs [hunk 5]
```diff
@@ -376,7 +376,7 @@ private BaseDirectory CreateDirectory(string indexName)
                 // Lucene is not thread safe on this call.
                 lock (_synLock)
                 {
-                    return FSDirectory.Open(path);
+                    return LStore.FSDirectory.Open(path);
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:379]:
```suggestion
                    return FSDirectory.Open(path);
```

File: src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs [hunk 6]
```diff
@@ -395,7 +395,7 @@ private async Task WriteAsync(string indexName, Action<IndexWriter> action, bool
                         var config = new IndexWriterConfig(LuceneSettings.DefaultVersion, analyzer)
                         {
                             OpenMode = OpenMode.CREATE_OR_APPEND,
-                            WriteLockTimeout = Lock.LOCK_POLL_INTERVAL * 3
+                            WriteLockTimeout = LStore.Lock.LOCK_POLL_INTERVAL * 3
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs:398]:
```suggestion
                            WriteLockTimeout = LuceneLock.LOCK_POLL_INTERVAL * 3
```

**Review by hishamco — CHANGES_REQUESTED** [review]

**Pull request merged**